src/transformStateWithClones.js

```diff
@@ -7,7 +7,35 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateHistory = [];
+  const stateCopy = { ...state };
+
+  for (const obj of actions) {
+    switch (obj.type) {
+      case 'addProperties':
+        for (const key in obj.extraData) {
+          stateCopy[key] = obj.extraData[key];
+        }
+        stateHistory.push({ ...stateCopy });
+        break;
+
+      case 'removeProperties':
+        for (const key of obj.keysToRemove) {
+          delete stateCopy[key];
+        }
+        stateHistory.push({ ...stateCopy });
```

Comment:
Similar to the previous comment, avoid mutating `stateCopy` directly. Create a new object that excludes the specified keys instead.

Reply:
as to a previous comment, that's a new object, I can change it